### ISSUE TITLE
Amplify push api: Maximum call stack size exceeded 

### ISSUE THREAD

**Issue by oliverandersencox** [aws-amplify/amplify-cli#4631]
>**Note**: If your issue/bug is regarding the AWS Amplify Console service, please log it in the 
[Amplify Console GitHub Issue Tracker](https://github.com/aws-amplify/amplify-console/issues)

**Describe the bug**
I have just updated the amplify and nw I cannot ush any API changes:

```
Maximum call stack size exceeded
There was an error pushing the API resource
RangeError: Maximum call stack size exceeded
```

**Amplify CLI Version**
4.21.3

**To Reproduce**
Add a new field to any model and run either:
- amplify push
- amplify push api

**Expected behavior**
A normal api push 

**Screenshots**

**Desktop (please complete the following information):**
- MAC OSX 10.15.4

I have ran a fresh amplify pull and also restarted the mac

**Comment by akshbhu** [comment]
Hi @oliverandersencox 

* Does your schema contains 25 or more model?
* Can you share the graphql schema which leads this error?

One potential workaround that might help is using amplify push --minify to minify your CFN templates.
Let us know if that works for you.

**Comment by oliverandersencox** [comment]
Yes it contains around 45 - I shall try that now

**Comment by akshbhu** [comment]
yes you can try that and let us know if that works.

I am closing this issue as a duplicate of #2584. You can follow updates and comment on this issue.

**Comment by oliverandersencox** [comment]
@akshbhu no luck - same error

**Comment by oliverandersencox** [comment]
@ashika01 same problem @akshbhu This isnt the same as that issue as I have also had that issue before.

I removed 3 tables from my model as they are us-used.

On top of this I have also been adding and removed fields from these models without error for the past month. This popped up today when I tried to add 1 field. Now with the attempt to remove 3 tables it still gives this issue.

I have a total of 32 tables FYI.

Happy to email my schema over

**Comment by akshbhu** [comment]
Hi @oliverandersencox 

I have reopened this issue. You can upload schema here or if it contains sensitive information you can email it to amplify-cli@amazon.com with this issue in the subject

**Comment by oliverandersencox** [comment]
@akshbhu thanks! Have emailed it across

**Comment by oliverandersencox** [comment]
@akshbhu any update on this? I never got a reply?

**Comment by oliverandersencox** [comment]
@akshbhu ???

**Comment by akshbhu** [comment]
Hi @oliverandersencox 

I am still working on this and will update you soon regarding this issue. I used your schemas to reproduce the issue and I got same error but not on amplify push but in updating the api.

**Comment by oliverandersencox** [comment]
@akshbhu okay great, thanks for the update

**Comment by oliverandersencox** [comment]
@akshbhu is there any news on this? We're blocked from release at the moment

**Comment by oliverandersencox** [comment]
any update on this guys?? @SwaySway @akshbhu

**Comment by oliverandersencox** [comment]
@akshbhu Being left it the dark now. Is there anything you can tell me?

I've contacted AWS support to increase the limit as advised in the other request - however they have said a limit increase wont affect this.

**Comment by oliverandersencox** [comment]
@SwaySway Hello!!!?

**Comment by SwaySway** [comment]
Hello @oliverandersencox 
I was able to reproduce the stack size error, I'll update the thread with my findings to see where this differs from i2584.

**Comment by oliverandersencox** [comment]
Thanks so much!

**Comment by SwaySway** [comment]
@oliverandersencox 
There seems to be a recursive type in one of your schema files.
In the `@auth` transformer we propagate auth directives to ObjectType nodes that are referenced from `@model`.
The transformer should be able to add the auth directives and not revisit the type if they have been added.

**Comment by oliverandersencox** [comment]
@SwaySway thanks for looking into it! I'm a bit unclear on what you mean, are you saying the issue will be in auth auth directive of one of the models?

**Comment by oliverandersencox** [comment]
@SwaySway this issues started when I added 1 attribute to a table though?

**Comment by SwaySway** [comment]
@oliverandersencox 

> are you saying the issue will be in auth auth directive of one of the models?
Correct using a similar example of what is causing the problem as noted below.
```
type Post @model
@auth(rules: [{ allow: owner }, { allow: private, operations: [read], provider: iam }])
{
  id: ID!
  title: String
  tags: [Tag]
}
# the transformer will propagate the auth directives such as @aws_iam @aws_cognito_user_pools
# down to the type tag
type Tag {
  id: ID!
  name: String
  # because of the type references itself it recursively goes through and adds the tags
  childTags: [Tag]
}
```
Was that change a type that referenced itself like example above? This is what I noticed caused the error from the provided schema.

**Comment by oliverandersencox** [comment]
@SwaySway - the change I made was a simple string attrbiute addition to a model. 

After reading you above post, I then removed all @aws_iam @aws_cognito_user_pools that I had one my types and i am still getting them same error - Can I send you the schema to look at?

We really need to update the api for our clients to use a new feature and this is blocking us!

**Comment by oliverandersencox** [comment]
@SwaySway - I removed all local amplify data and ran amplify pull - then added one string attribute after the updated amplify cli - same issue

**Comment by mdlavin** [comment]
Thank you for putting this PR up. I pulled it in my own branch and it resolved the `Maximum call stack size exceeded` for me

**Comment by SwaySway** [comment]
Closing this issue as the fix for this is now included in the v4.24.1. Should you still experience issues with this please comment below.
